asgs_main.sh

```diff
@@ -2118,11 +2118,27 @@ if [[ $START = coldstart ]]; then
    logMessage "$ENSTORM: $THIS: Constructing control file with the following options: $CONTROLOPTIONS."
 
 #BOB
-   echo "Debug: hindcast: building fort.15" >> ${SYSLOG} 2>&1
+   logMessage "Debug: hindcast: building fort.15"
+   controlFile="$ADVISDIR/$ENSTORM/fort.15"
+   swanFile="$ADVISDIR/$ENSTORM/fort.26"
    perl $SCRIPTDIR/control_file_gen.pl $CONTROLOPTIONS >> ${SYSLOG} 2>&1
-   if [ ! -s "$ADVISDIR/$ENSTORM/fort.15" ] ; then
-      echo "hindcast: $ADVISORDIR/$ENSTORM/fort.15 file is 0-length.  This is terminal."  >> ${SYSLOG} 2>&1
-      RMQMessage "EXIT" "$CURRENT_EVENT" "$THIS" "FAIL" "hindcast: fort.15 file is 0-length.  This is terminal."
+   controlExitStatus=$?
+   if [[ $controlExitStatus != 0 ]]; then
+      controlMsg="The control_file_gen.pl script failed with the following error code: '$controlExitStatus'."
+   fi
+   if [[ ! -e $controlFile || ! -s $controlFile ]]; then
+      controlExitStatus=1
```

Comment:
What is the purpose of changing this `controlExitStatus`? Is this just an error code?

Reply:
`controlExitStatus` is kind of doing double duty. The actual exit status is used to construct the initial error message, if the exit status is other than zero. After that, `controlExitStatus` is just a flag to indicate whether the perl script exited with an error. The purpose is to maintain troubleshooting granularity between the perl script exiting with an error, and any issues with the resulting `fort.15` and/or `fort.26`. Open to suggestions on improving this.

Reply:
I think it is working ok, although it could be improved.